<?xml version="1.0"?>
<launch>

  <arg name="map_file_path" default="/home/sun/Pictures/maps"/>
  <!-- 保存地图的位置 -->
  <arg name="map_file_base_name" default="Explorer_geotiff_mapper_two"/>
  <!-- 保存地图的名字 -->
  <arg name="team_name" default="Explorer"/>
  <arg name="country" default="China"/>
  <arg name="mission_name" default="MISSION"/>
  
  
  <!-- 打开hector_geotiff节点 -->
  <node pkg="hector_geotiff" type="geotiff_node" name="hector_geotiff_node" output="screen" launch-prefix="nice -n 15">
  
    <!--xterm 命令表示新建一个terminal； -e参数告诉xterm执行剩下的命令行。-->
    <param name="map_file_path" type="string" value="$(arg map_file_path)" />
    <param name="map_file_base_name" type="string" value="$(arg map_file_base_name)" />
   
    <param name="geotiff_save_period" type="double" value="5" />
    <!--每多长时间保存一次地图　单位是秒-->
    
    <!--2018.1.13-->
    <param name="draw_background_checkerboard" type="bool" value="true" />
    <param name="draw_free_space_grid" type="bool" value="true" /> 
    <!--2018.1.13-->

    <param name="plugins" type="string" value="hector_geotiff_plugins/TrajectoryMapWriter hector_worldmodel_geotiff_plugins/VictimMapWriter hector_worldmodel_geotiff_plugins/QRCodeMapWriter" />

    <param name="VictimMapWriter/draw_all_objects" value="true" />
    <param name="VictimMapWriter/class_id" value="victim" />
    <param name="VictimMapWriter/service_add_victim" value="srv_add_victim"/>
    <param name="VictimMapWriter/team_name"   value="$(arg team_name)"/>
    <param name="VictimMapWriter/country"   value="$(arg country)"/>
    <param name="VictimMapWriter/mission_name"   value="$(arg mission_name)"/>
    <param name="VictimMapWriter/same_victim_torlerence" value="0.6"/>

    <param name="QRCodeMapWriter/draw_all_objects" value="true" />
    <param name="QRCodeMapWriter/class_id" value="qrcode" />
    <param name="QRCodeMapWriter/team_name"   value="$(arg team_name)"/>
    <param name="QRCodeMapWriter/country"   value="$(arg country)"/>
    <param name="QRCodeMapWriter/mission_name"   value="$(arg mission_name)"/>
    <!--调用插件 并设置插件的参数-->

  </node><!--end of hector_geotiff-->


<!-- 打开这个记录路径的节点 -->
<!-- <arg name="trajectory_source_frame_name" default="/base_link"/>
<arg name="trajectory_update_rate" default="4"/>
<arg name="trajectory_publish_rate" default="0.25"/>

<node pkg="hector_trajectory_server" type="hector_trajectory_server" name="hector_trajectory_server" output="screen">

  <param name="target_frame_name" type="string" value="/map" />
  <param name="source_frame_name" type="string" value="$(arg trajectory_source_frame_name)" />
  <param name="trajectory_update_rate" type="double" value="$(arg trajectory_update_rate)" />
  <param name="trajectory_publish_rate" type="double" value="$(arg trajectory_publish_rate)" />
</node> -->


</launch>


